cmake/projects/Boost/schemes/url_sha1_boost_library.cmake.in

```diff
@@ -286,6 +288,65 @@ file(
     "${using_mpi}\n"
 )
 
+# *** handle requested python version if such is specified ***
+#
+hunter_parse_cmake_args_for_keyword(
+  CMAKE_ARGS "@HUNTER_Boost_CMAKE_ARGS@"
+  KEYWORD REQUESTED_PYTHON_VERSION
+  OUTPUT requested_python
+)
+# determine paths required to set up python in boost.user.jam
+if(requested_python)
+  #FindPythonInterp has been deprecated since CMake 3.12 in favor of
```

Comment:
Any benefits of using two branches here? I mean if we can use old `find_package(Python ${requested_python} EXACT QUIET COMPONENTS Interpreter Development)` way for both why we need `find_package(PythonInterp ${requested_python} EXACT QUIET)`?

Reply:
Because ``find_package(Python ... `` isn't supported on versions of CMake before 3.12, while ``find_package(PythonInterp`` is deprecated since 3.12.

Reply:
> while find_package(PythonInterp is deprecated since 3.12

Deprecated is not mean that we can't use it.

I think we have two choices here.

First is to make 3.12 a requirement for this part of code:
```cmake
if(CMAKE_VERSION VERSION_LESS "3.12")
  hunter_user_error("...")
endif()
```
Meaning if user want to add `REQUESTED_PYTHON_VERSION=<VALUE>` then CMake 3.12+ should be used, otherwise standard CMake 3.2+ is a requirement.

Second is to use same code, i.e. remove current `if(CMAKE_VERSION VERSION_LESS 3.12)` block.

Having two branches look suspicious to me. In theory it mean that having same configuration, but running it with different CMake versions can produce different artifacts, that's violation of Hunter design.

Reply:
Хозяин - барин, как говориться, and I'll remove the branch for the newer versions of CMake, but "Deprecated" means "can, but should not, be used, and will very likely be removed altogether in some future release". You'll have to fix it then when it breaks in the future. I need this to work for all CMake versions >= 3.5.

Reply:
> but "Deprecated" means "can, but should not, be used

As I said there are two ways to proceed:
* Follow rules pedantically: deprecated code should not be used - it means make CMake 3.12 a requirement
* Relax rules for practical reason: use deprecated code - have same code for all CMake versions

Your code is somewhere in between, not following any principle, not fixing issues, make code hairy and add potential instability to system. Doesn't sound good for me.

> will very likely be removed altogether in some future release

I doubt it will be removed in observable future (probably in CMake 4) because it will break [quite a lot of CMakeLists.txt files](https://github.com/search?l=&q=find_package+PythonInterp+language%3ACMake+language%3ACMake&type=Code). Even `exec_program` from [CMake 2.6](https://cmake.org/cmake/help/cmake2.6docs.html#command:exec_program) is still working fine :)